### ISSUE TITLE
ERROR: Could not find a version that satisfies the requirement tensorflow==2.17.0 

### ISSUE THREAD

**Issue by tqx94** [aiverify-foundation/moonshot#304]
Hi,  

I am using MacOS, my python version is 3.11.

I get the following error when running the command: python -m moonshot -i moonshot-data -i moonshot-ui
I also get the same error when I pip install requirements.txt from moonshot-data folder directly.
When I tried to upgrade my tensorflow(current version 2.16) to tensorflow 2.17, i get this error too.  

```
Requirement already satisfied: tensorflow-io-gcs-filesystem==0.37.1 in /Users/ncsadmin/path/to/venv/lib/python3.11/site-packages (from -r requirements.txt (line 164)) (0.37.1)
ERROR: Ignored the following versions that require a different python version: 1.6.2 Requires-Python >=3.7,<3.10; 1.6.3 Requires-Python >=3.7,<3.10; 1.7.0 Requires-Python >=3.7,<3.10; 1.7.1 Requires-Python >=3.7,<3.10; 1.7.2 Requires-Python >=3.7,<3.11; 1.7.3 Requires-Python >=3.7,<3.11; 1.8.0 Requires-Python >=3.8,<3.11; 1.8.0rc1 Requires-Python >=3.8,<3.11; 1.8.0rc2 Requires-Python >=3.8,<3.11; 1.8.0rc3 Requires-Python >=3.8,<3.11; 1.8.0rc4 Requires-Python >=3.8,<3.11; 1.8.1 Requires-Python >=3.8,<3.11
ERROR: Could not find a version that satisfies the requirement tensorflow==2.17.0 (from versions: 2.12.0rc0, 2.12.0rc1, 2.12.0, 2.12.1, 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0, 2.13.1, 2.14.0rc0, 2.14.0rc1, 2.14.0, 2.14.1, 2.15.0rc0, 2.15.0rc1, 2.15.0, 2.15.1, 2.16.0rc0, 2.16.1, 2.16.2)
ERROR: No matching distribution found for tensorflow==2.17.0
```

Appreciate if you could let me know if there is any way to install resolve this?

Please find some of the packages versions related to tensorflow below. 

```
tensorboard==2.16.2
tensorboard-data-server==0.7.2
tensorflow==2.16.2
tensorflow-io-gcs-filesystem==0.37.1
termcolor==2.4.0
text-unidecode==1.3
```

**Comment by imda-normanchia** [comment]
hello,

to better help solve your issue, would you mind sharing what's your pip version. you. can do so by running `pip --version`

**Comment by Musyaffaq** [comment]
Hi @imda-normanchia,

I'm facing a similar issue. The pip version I'm using is 24.2

**Comment by imda-normanchia** [comment]
Hello @Musyaffaq, have you tried reinstalling moonshot in a new virtual environment? another solution you can try is to uninstall the tensorflow dependency in our environment and try to reinstall again as there is a version discrepancy between the version of tensorflow we required and what your system have.

**Comment by Musyaffaq** [comment]
Hi @imda-normanchia, I installed it in a new environment (I'm not OP btw). I have tested it on both a Intel and M1 Macbook, both gives me the same error as OP:

`ERROR: Ignored the following versions that require a different python version: 0.28.0 Requires-Python >=3.7, <3.11; 1.6.2 Requires-Python >=3.7,<3.10; 1.6.3 Requires-Python >=3.7,<3.10; 1.7.0 Requires-Python >=3.7,<3.10; 1.7.1 Requires-Python >=3.7,<3.10; 1.7.2 Requires-Python >=3.7,<3.11; 1.7.3 Requires-Python >=3.7,<3.11; 1.8.0 Requires-Python >=3.8,<3.11; 1.8.0rc1 Requires-Python >=3.8,<3.11; 1.8.0rc2 Requires-Python >=3.8,<3.11; 1.8.0rc3 Requires-Python >=3.8,<3.11; 1.8.0rc4 Requires-Python >=3.8,<3.11; 1.8.1 Requires-Python >=3.8,<3.11
ERROR: Could not find a version that satisfies the requirement tensorflow==2.17.0 (from versions: 2.12.0rc0, 2.12.0rc1, 2.12.0, 2.12.1, 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0, 2.13.1, 2.14.0rc0, 2.14.0rc1, 2.14.0, 2.14.1, 2.15.0rc0, 2.15.0rc1, 2.15.0, 2.15.1, 2.16.0rc0, 2.16.1, 2.16.2)
ERROR: No matching distribution found for tensorflow==2.17.0`

I suspect that tensorflow version 2.17.0 is not available for macOS? Because when I pip install tensorflow, it installs version 2.16.2 instead.

**Comment by imda-normanchia** [comment]
@Musyaffaq  hi, i am using M2 Macbook pro and I do have tensorflow 2.17.0 installed on my system. 

my pip version is 24.2 and python version is 3.11.0

may i know if we are using similar python version? 

and also are you using virtual environment or conda to do the installation?

**Comment by Musyaffaq** [comment]
Hi @imda-normanchia, yes everything same for me, but I was using conda environment. However, I figured out why it was not working (on the Intel Macbook).

Essentially, the packages tensorflow, torch and torchvision no longer supports for macOS x86 builds after version 2.16, 2.2 and 0.17 respectively. So I have modified the moonshot-data/requirements.txt file to the versions available for the Intel Macbook. I had to modify some other packages which depended on an older version of the above packages. Here it is for your reference:

ml-dtypes~=0.3.1
tensorboard~=2.16.0
tensorboard~=2.16.0
tf-keras~=2.16.0
torch~=2.2.2
torchvision~=0.17.2

So far I am able to use the tool with no issue. Thanks for your response anyway!
